presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ScalarAggregationToJoinRewriter.java

```diff
@@ -165,10 +168,25 @@ private PlanNode rewriteScalarAggregation(
                     assignments);
         }
         else {
-            return aggregationNode.get();
+            Assignments assignments = Assignments.builder()
+                    .putAll(Assignments.identity(aggregationOutputSymbols))
+                    .build();
+
+            return new ProjectNode(
+                    idAllocator.getNextId(),
+                    aggregationNode.get(),
+                    assignments);
         }
     }
 
+    private static List<Symbol> getTruncatedAggregationSymbols(ApplyNode applyNode, AggregationNode aggregationNode)
+    {
+        Set<Symbol> applySymbols = new HashSet<>(applyNode.getOutputSymbols());
+        return aggregationNode.getOutputSymbols().stream()
```

Comment:
What if `AggregationNode` does not produce some symbol from `ApplyNode`'s output symbols (e.g. due to an error in rewrite)? With your approach, optimizer's sanity check wouldn't catch that